xc7/primitives/slicem/Ndram/ntemplate.N_dram.pb_type.xml

```diff
@@ -70,22 +70,23 @@
     </metadata>
   </mode>
   <mode name="64_SINGLE_PORT">
-    <xi:include href="spram64.pb_type.xml" />
+    <xi:include href="dpram64.pb_type.xml" />
     <metadata>
       <meta name="fasm_features">
         DI1MUX.{N}I
       </meta>
     </metadata>
     <interconnect>
-      <direct name="CLK" input="{N}_DRAM.CLK" output="SPRAM64.CLK" />
-      <direct name="A" input="{N}_DRAM.A" output="SPRAM64.A" />
-      <direct name="WA7" input="{N}_DRAM.WA[6]" output="SPRAM64.WA7" />
-      <direct name="WA8" input="{N}_DRAM.WA[7]" output="SPRAM64.WA8" />
-      <direct name="{N}I" input="{N}_DRAM.{N}I" output="SPRAM64.DI1" />
-      <direct name="WE" input="{N}_DRAM.WE" output="SPRAM64.WE" />
+      <direct name="CLK" input="{N}_DRAM.CLK" output="DPRAM64.CLK" />
+      <direct name="A" input="{N}_DRAM.A" output="DPRAM64.A" />
+      <direct name="WA" input="{N}_DRAM.A[5:0]" output="DPRAM64.WA[5:0]" />
```

Comment:
This is not right, other than DLUT, all DRAM's get their address lines from the D1-D6 connections.

Reply:
This is in single port mode, where the read and write ports are tied together, just like in DLUT.

Reply:
> This is in single port mode, where the read and write ports are tied together, just like in DLUT.

Have we confirmed this is correct? Specifically have we tested whether Vivado will pack two RAM64X1D's into the same slice if their WA lines vary?  I believe we ran that test and it showed that Vivado would NOT pack those together, indicated that WA is always from the DLUT input pins.

Reply:
I think you're right from looking at the CLB User Guide. I'll check Vivado.

Reply:
WA7/WA8/WE are also slice-wide, so they should be moved out into slicem.pb_type.xml

Reply:
https://github.com/SymbiFlow/symbiflow-arch-defs/blob/dc45861228a81ade14f22d828856a575f9b471d9/xc7/primitives/slicem/slicem.pb_type.xml#L267-L288

It's just handled in slicem.pb_type.xml. That's probably the easiest to understand way of doing it.

Reply:
I've checked that Vivado doesn't pack RAMs with independent data inputs together.